arbitrum-docs/stylus/concepts/stylus-cache-manager.md

```diff
@@ -0,0 +1,33 @@
+---
+title: 'Stylus caching strategy'
+description: 'A conceptual overview of the Stylus caching strategy and CacheManager contract, and explaining its functionality.'
+sme: mahsa-moosavi
+target_audience: 'Developers deploying smart contracts using Stylus.'
+sidebar_position: 3
+---
+
+Stylus is designed for fast computation, but it faces significant performance issues due to the slow activation process when entering a contract. This slow [activation](https://docs.arbitrum.io/stylus/stylus-gentle-introduction#activation) takes up substantial resources and time. These repeated activations lead to inefficiencies and hinder the overall performance of the system. To address this problem, we have implemented a caching strategy. The cache serves as a bypass, storing frequently accessed contracts in-memory. This approach allows for avoiding repeated activations, thereby saving resources and time while significantly improving the speed and efficiency of contract execution.
+
+import PublicPreviewBannerPartial from '../partials/_stylus-public-preview-banner-partial.md';
+
+<PublicPreviewBannerPartial />
+
+## CacheManager contract
+
+The core component of our caching strategy is the [CacheManager contract](https://github.com/OffchainLabs/stylus-contracts/blob/c51ab1dc90f543caf579600162f77e053984b8cd/src/chain/CacheManager.sol). This smart contract manages the cache, interacts with precompiles, and determines which contracts should be cached. The CacheManager can hold approximately 4,000 contracts in memory.
+
+The CacheManager defines how contracts remain in the cache and how they compete with other contracts for cache space. Its primary purpose is to manage high initialization costs, ensuring efficient contract activation and usage. The contract includes methods for adding and removing cache entries, querying the status of cached contracts, and managing the lifecycle of cached data.
+
+### Key features
+
+The CacheManager plays a crucial role in our caching strategy by keeping a certain set of contracts in-memory rather than retrieving them from disk. This significantly reduces the activation time for frequently accessed contracts. The CacheManager contract is an on-chain contract that accepts bids for inserting contract code into the cache. It then calls a precompile that loads or unloads the contracts in the ArbOS cache, which follows the on-chain cache but operates locally in the client, and marks the contract as in or out of the cache in the ArbOS state.
```

Comment:
Is it true that the cache manager reduces only activation time? Wouldn't subsequent calls (after activation) be cheaper in gas too?